### PR TITLE
=clu #13802 Introduce max-total-nr-of-instances for cluster aware routers

### PR DESCRIPTION
**Pull request by patriknw**
This was rather complicated to figure out, but since tests were failing when I did it wrong I think we have good test coverage.

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #13802

File: akka-cluster/src/main/scala/akka/cluster/ClusterActorRefProvider.scala [hunk 1]
```diff
@@ -60,9 +61,19 @@ private[akka] class ClusterActorRefProvider(
...
   override def parseConfig(path: String, config: Config): Option[Deploy] = {
+    // config is the user supplied section, no defaults
+    // amend it to use max-total-nr-of-instances as nr-of-instances if cluster.enabled and
+    // user has not specified nr-of-instances
+    val config2 =
+      if (config.hasPath("cluster.enabled") && config.getBoolean("cluster.enabled") && !config.hasPath("nr-of-instances")) {
+        val maxTotalNrOfInstances = config.withFallback(default).getInt("cluster.max-total-nr-of-instances")
+        ConfigFactory.parseString("nr-of-instances=" + maxTotalNrOfInstances)
+          .withFallback(config)
+      } else config
```
Comment (ktoso) [inline akka-cluster/src/main/scala/akka/cluster/ClusterActorRefProvider.scala]: that's the complicated bit hm

File: akka-docs/rst/scala/cluster-usage.rst [hunk 2]
```diff
@@ -464,8 +464,9 @@ actors to which the messages will be forwarded to by the router.
 Messages will be forwarded to the routees using :ref:`ActorSelection <actorSelection-scala>`, so the same delivery semantics should be expected.
 It is possible to limit the lookup of routees to member nodes tagged with a certain role by specifying ``use-role``.
 
-``nr-of-instances`` defines total number of routees in the cluster. Setting ``nr-of-instances`` 
-to a high value will result in new routees added to the router when nodes join the cluster.
+``max-total-nr-of-instances`` defines total number of routees in the cluster. By default ``max-total-nr-of-instances`` 
+is set to a high value (10000) that will result in new routees added to the router when nodes join the cluster.
```
Comment (ktoso) [inline akka-docs/rst/scala/cluster-usage.rst]: `which` will result in

Comment (ktoso) [inline akka-docs/rst/scala/cluster-usage.rst]: actually, nevermind

**Comment by ktoso** [comment]
LGTM afaics, needs a rebase

**Pull request merged**